darts/models/forecasting_model.py

```diff
@@ -101,6 +105,343 @@ def _build_forecast_series(self,
 
         return TimeSeries.from_times_and_values(time_index, points_preds, freq=self.training_series.freq())
 
+    def _backtest_sanity_checks(self, *args: Any, **kwargs: Any) -> None:
+        """Sanity checks for the backtest function
+
+        Parameters
+        ----------
+        args
+            The args parameter(s) provided to the backtest function.
+        kwargs
+            The kwargs paramter(s) provided to the backtest function.
+
+        Raises
+        ------
+        ValueError
+            when a check on the parameter does not pass.
+        """
+        # parse args and kwargs
+        if len(args) > 0:
+            training_series = args[0]
+        else:
+            training_series = kwargs['training_series']
+        n = SimpleNamespace(**kwargs)
+
+        # check target and training series
+        if not hasattr(n, 'target_series'):
+            target_series = training_series
+        else:
+            target_series = n.target_series
+        raise_if_not(all(training_series.time_index() == target_series.time_index()), "the target and training series"
+                     " must have the same time indices.")
+
+        # check forecast horizon‚
+        if hasattr(n, 'forecast_horizon'):
+            raise_if_not(n.forecast_horizon > 0, 'The provided forecasting horizon must be a positive integer.', logger)
+
+        # check start parameter
+        if hasattr(n, 'start'):
+            if isinstance(n.start, float):
+                raise_if_not(n.start >= 0.0 and n.start < 1.0, '`start` should be between 0.0 and 1.0.', logger)
+            elif isinstance(n.start, pd.Timestamp):
+                raise_if_not(n.start in training_series, '`start` timestamp is not in the `series`.', logger)
+                raise_if(n.start == training_series.end_time(), '`start` timestamp is the last timestamp of `series`',
+                         logger)
+            else:
+                raise_if(training_series[n.start].time_index()[0] == training_series.end_time(), '`start` timestamp '
```

Comment:
We could also add an additional check that `start + forecast_horizon` is in the series if `trim_to_series` is False.

Reply:
> Instead of doing `.time_index()[0]` you should be able to call `.start_time()`.
> Also, shouldn't we replace the `==` by `>=`, and add a preliminary check that (when it's an int), it is >= 0.

done

Reply:
> We could also add an additional check that `start + forecast_horizon` is in the series if `trim_to_series` is False.

I assume what you meant to say is if `trim_to_series` is **True**', since otherwise we can have points in our predicted series that go beyond the input series time index, right?

Reply:
Right, correct